packages/hardhat-core/src/internal/core/tasks/task-definitions.ts

```diff
@@ -560,6 +560,14 @@ export class OverriddenTaskDefinition implements TaskDefinition {
     this.parentTaskDefinition = parentTaskDefinition;
   }
 
+  public toJSON() {
+    return {
+      name: this.name,
+      isSubtask: this.isSubtask,
+      paramDefinitions: this.paramDefinitions,
+    };
+  }
```

Comment:
The root of the problem is that, for an `OverridenTaskDefinition`, the `name` field exists but, since it's a getter, it's not serialized.

An obvious solution is to remove those getters (and be overly cautious with getters in the future).

A more type-safe (but complex) solution is to add a `SerializableTaskDefinition` interface, extend that from `TaskDefinition` and use it in the `autocomplete` module, and then return that type here in the `toJSON` method.